libraries/ESP8266WiFi/src/ESP8266WiFiType.h

```diff
@@ -48,6 +48,13 @@ typedef enum WiFiSleepType
     WIFI_NONE_SLEEP = 0, WIFI_LIGHT_SLEEP = 1, WIFI_MODEM_SLEEP = 2
 } WiFiSleepType_t;
 
+// ESP32 compatibility
+typedef enum wifi_ps_type
+{
+    WIFI_PS_NONE = WIFI_NONE_SLEEP,
+    WIFI_PS_MIN_MODEM = WIFI_LIGHT_SLEEP,
+    WIFI_PS_MAX_MODEM = WIFI_MODEM_SLEEP,

```

Comment:
Is this right though? MODEM is default, hence it should be MIN? LIGHT is 'sleepier' mode afaik

Reply:
See this comment:
https://github.com/esp8266/Arduino/pull/7901#discussion_r584594164

Reply:
> From the docs on ESP8266 it seems that WIFI_PS_MIN_MODEM == WIFI_LIGHT_SLEEP and WIFI_PS_NONE == WIFI_NONE_SLEEP

https://www.espressif.com/sites/default/files/documentation/2c-esp8266_non_os_sdk_api_reference_en.pdf ?
(still applies to the 2.2.x branch)
> 3.5.54. wifi_set_sleep_type 
> - Default mode: Modem-sleep.
-  In order to lower the power comsumption, ESP8266 changes the TCP timer
tick from 250ms to 3s in light-sleep mode, which leads to increased timeout for
TCP timer. Therefore, the modem-sleep or deep-sleep mode should be used
where there is a requirement for the accurancy of the TCP timer.

Plus:
https://github.com/espressif/arduino-esp32/blob/15bae92a72c102331e7eaf7dcb439e8ad6920184/tools/sdk/include/esp32/esp_wifi_types.h#L192-L196

Meaning, NONE - power-save features are disabled, MIN - minimum amount of power-saving features, but some are enabled, MAX - maximum power-save mode

Reply:
Then I don't understand the number 3 in @d-a-v comment. Why `WIFI_MODEM_SLEEP` is on enabled and `WIFI_LIGHT_SLEEP` is on disabled?

Reply:
:shrug:
Following esp32 api, 1st choice is used there. ::setSleep(enum) is used to select the MAX sleep mode
(ref. https://github.com/espressif/arduino-esp32/issues/3896 and https://github.com/espressif/arduino-esp32/issues/3900)

Reply:
I was wrong, sorry! I swapped MODEM and LIGHT sleep modes.
MODEM is indeed the default and is between NONE and LIGHT.